scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1466,143 +1470,174 @@
-        (.addContainersItem pod-spec container)))
+      (let [get-resource-requirements-fn (fn [fieldname] (if use-cook-sidecar?
+                                                           (get-in sidecar [:resource-requirements fieldname])
+                                                           0))
+            total-memory-request (add-as-decimals computed-mem (get-resource-requirements-fn :memory-request))
+            total-memory-limit (add-as-decimals computed-mem (get-resource-requirements-fn :memory-limit))
+            total-cpu-request (add-as-decimals cpus (get-resource-requirements-fn :cpu-request))
+            total-cpu-limit (add-as-decimals cpus (get-resource-requirements-fn :cpu-limit))]
+        ;init container
+        (when use-cook-init?
+          (let [{:keys [command image image-fallback]} init-container
+                container (V1Container.)
+                resources (V1ResourceRequirements.)]
+            ; container
+            (.setName container cook-init-container-name)
+            (.setImage container (resolve-image-from-incremental-config
+                                   job passport-event-base passport/init-container-image-selected
+                                   image image-fallback))
+            (.setCommand container command)
+            ; Shares the same workdir as the checkpoint container.
+            (.setWorkingDir container init-container-workdir)
+            (.setEnv container main-env-vars)
+            (.setVolumeMounts container (filterv some? (concat [(init-container-workdir-volume-mount-fn false)
+                                                                (scratch-space-volume-mount-fn false)]
+                                                               init-container-checkpoint-volume-mounts)))
+            (set-mem-cpu-resources resources total-memory-request total-memory-limit total-cpu-request total-cpu-limit)
+            (.setResources container resources)
+            (.addInitContainersItem pod-spec container)))
+
+        ; checkpoint injection container
+        ; This container is used for injecting checkpoint/restore functionality into the user's main container.
+        ;
+        ; This is run in the same image as the user's container, but run with an alternate command line. The reason is that
+        ; the init container may not use the same platform image as the user's container, so certain binaries or other
+        ; images can't be copied between those container.
+        ;
+        ; The idea is the init container injects some scripts into a shared volume used by the checkpoint container.
+        ; The checkpoint container then runs. As it runs with the same image as the user container, those scripts can
+        ; inspect the platform the user container is on and install the right checkpoint or other code.
+        (when use-checkpoint-injection?
+          (let [{:keys [command]} checkpoint-container
```

Comment:
So, the k8s config will have a {"checkpoint-container" {"command" "<checkpoint command>"}} entry? We don't need to do something like {"<image name>" "<checkpoint command for image"}, right?

Reply:
Yes. Just a command.

No. Remember the comment above: We're using the user's container's image, so that doesn't need to be specified.